src/pass.h

```diff
@@ -88,14 +88,28 @@ struct PassRunner {
 //
 // Core pass class
 //
-class Pass : public WasmWalker {
- public:
-  // Override this to perform preparation work before the pass runs
+class Pass {
+public:
+  virtual ~Pass() {};
```

Comment:
If I do this, compilation fails with a linker error. I suppose this is beacause it's not implemented in any of the derived classes.

```
Undefined symbols for architecture x86_64:
  "wasm::Pass::~Pass()", referenced from:
      wasm::MergeBlocks::~MergeBlocks() in MergeBlocks.cpp.o
      wasm::MergeBlocks::~MergeBlocks() in MergeBlocks.cpp.o
      non-virtual thunk to wasm::MergeBlocks::~MergeBlocks() in MergeBlocks.cpp.o
      non-virtual thunk to wasm::MergeBlocks::~MergeBlocks() in MergeBlocks.cpp.o
      wasm::RemoveUnusedBrs::~RemoveUnusedBrs() in RemoveUnusedBrs.cpp.o
      wasm::RemoveUnusedBrs::~RemoveUnusedBrs() in RemoveUnusedBrs.cpp.o
      non-virtual thunk to wasm::RemoveUnusedBrs::~RemoveUnusedBrs() in RemoveUnusedBrs.cpp.o
      ...
```